kimchi/src/circuits/constraints.rs

```diff
@@ -147,6 +155,48 @@ pub struct Builder<F: PrimeField> {
     lookup_tables: Vec<LookupTable<F>>,
     runtime_tables: Option<Vec<RuntimeTableCfg<F>>>,
     precomputations: Option<Arc<DomainConstantEvaluations<F>>>,
+    foreign_field_modulus: Vec<F>,
+}
+
+/// Create selector polynomial for a circuit gate
+pub fn selector_polynomial<F: PrimeField>(
```

Comment:
so you can delete these functions in `range_check` module now?

Reply:
actually, I think we could have a trait implemented for a generic constant so that the selector_polynomials function returned an array of length the number of polynomials for each gadget, meaning 4 for ChaCha, or 2 for RangeCheck for example. Otherwise the selector polynomial function inside constraints returns a vector. What do you think? Perhaps this should be addressed in a further PR that collects functionalities of gates and creates a larger trait for them.

Reply:
Ok, I realised this is not necessary and would over complicate things. Answering your question, yes, now these functions can be removed in range_check module, but also FFAdd and FFMul.